### ISSUE TITLE
Feature Request: Disables self invite

### ISSUE THREAD

**Issue by brunoarueira** [scambra/devise_invitable#836]
On the project I'm working on, the users will be only invited after certain operations (e.g. create an user record or invite manually through an authenticated area)

I thought this will be a good addition to the project with a simple option like `self_invite: false` (probably on the model like the `invite_for`, enabling multiple resources using devise to be enable or disable invitations with this new option) and this will disable the route.

A workaround I've made is to override the controller using the default devise mechanism and override the new action with a redirect to a root path.

Would you accept a pull request for that?

**Comment by scambra** [comment]
Sorry, I'm not sure if I understand your suggestion.

Do you want to use devise invitable to call invite! class method only from some custom controllers, or model callbacks, so you want to disable new and create actions in invitations controller? I can accept a pull request for that.

**Comment by scambra** [comment]
I think another way to do it is defining authenticate_inviter! to raise Exception, redirect or render an error, but if you add that option it could be a cleaner way

**Comment by brunoarueira** [comment]
Yeah, the suggested option will disable actions (`new` and `create`) on the invitations controller and avoid any unwanted user to invite himself.

**Comment by scambra** [comment]
Do you think adding authenticate_inviter! method to application controller is good enough, or do you want to send pull request?

If that's enough, you could improve README or add something to wiki, to help others

**Comment by brunoarueira** [comment]
I'll think a bit and return, because at my use case, the invited user needs to be invited by someone which previously sets some parameters and/or relationships.

**Comment by ukolovda** [comment]
I define `InviteController` and rewrote the method `invite_resource`:

```ruby
class Users::InvitationsController < Devise::InvitationsController

  private
  def invite_resource
    super do |new_user|
      if new_user == current_user
        new_user.errors.add :email, :taken
      end
    end
  end
end
```
